src/validators/module/validateLoader.js

```diff
@@ -0,0 +1,93 @@
+import {SEPERATOR} from '../../constants'
+import {friendlyTypeOf} from '../../utils'
+import _ from 'lodash'
+
+/**
+ *  A loader is an object which can have the following props:
+ *  (from https://webpack.github.io/docs/configuration.html#module-loaders)
+ *
+ *   test: A condition that must be met
+ *   exclude: A condition that must not be met
+ *   include: A condition that must be met
+ *   loader: A string of “!” separated loaders
+ *   loaders: An array of loaders as string
+ *
+ *   A condition may be a RegExp (tested against absolute path),
+ *   a string containing the absolute path,
+ *   a function(absPath): bool, or an array of one of these combined with “and”.
+ */
+
+export function formatErrorMessage(loader, messages) {
+  return `While looking at loader ${JSON.stringify(loader)}, ` +
+         `the following errors were found: ${messages.join(SEPERATOR)}`
+}
+
+// Splitting up this function into more granular parts would
+// increase cognitive load even more, so disabling cyclomatic complexity check here.
+/* eslint-disable complexity */
+function validateLoaderAndLoadersProperty({loader, loaders, query}) {
+  const messages = []
+
+  if (!loader && !loaders) {
+    messages.push(`Found neither \`loader\` nor \`loaders\``)
+  }
+
+  if (loader && loaders) {
+    messages.push(`Found \`loader\` and \`loaders\`. These properties are mutually exclusive`)
+  }
+
+  if (loaders && query) {
+    messages.push(`You can only pass the \`query\` property when you have one \`loader\``)
+  }
+
+  if (loader && friendlyTypeOf(loader) !== 'string') {
+    messages.push(`Expected \`loader\` property to be a string, but got ${JSON.stringify(loader)}`)
```

Comment:
Well a loader object JSON won't be that big. The longest could be the loader string, maybe 80-100 chars. But I'm a fan of failing verbosely and I think it'll be an UX benefit to see which loader is faulty at first sight.